encoding/protobuf/parse.go

```diff
@@ -271,32 +271,45 @@ func (p *protoConverter) resolve(pos scanner.Position, name string, options []*p
 }
 
 func (p *protoConverter) resolveTopScope(pos scanner.Position, name string, options []*proto.Option) ast.Expr {
+	pkgPrefix := p.protoPkg + "."
 	for i := 0; i < len(name); i++ {
 		k := strings.IndexByte(name[i:], '.')
 		i += k
 		if k == -1 {
 			i = len(name)
 		}
-		if m, ok := p.scope[0][name[:i]]; ok {
-			if m.pkg != nil {
-				p.imported[m.pkg.qualifiedImportPath()] = true
+		curName := name[:i]
+		if strings.HasPrefix(curName, pkgPrefix) {
+			shortName := strings.Replace(curName, pkgPrefix, "", -1)
+			if m, ok := p.scope[0][shortName]; ok {
+				return p.resolveTopScopeHelper(pos, m.cue, m.pkg, name, i)
 			}
-			expr := m.cue()
-			for i < len(name) {
-				name = name[i+1:]
-				if i = strings.IndexByte(name, '.'); i == -1 {
-					i = len(name)
-				}
-				expr = ast.NewSel(expr, "#"+name[:i])
-			}
-			ast.SetPos(expr, p.toCUEPos(pos))
-			return expr
 		}
+		if m, ok := p.scope[0][curName]; ok {
+			return p.resolveTopScopeHelper(pos, m.cue, m.pkg, name, i)
+		}
```

Comment:
you don't need such a large refactor. you can just replace the original line

```
if m, ok := p.scope[0][name[:i]]; ok {
```

with something like

```
curName := name[:i]
if local, ok := strings.CutPrefix(curName, p.protoPkg + "."); ok {
    curName = local
}
if m, ok := p.scope[0][curName]; ok {
```

and you also avoid needing a new method.

Reply:
Fixed